sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -69,7 +69,8 @@ void gnrc_lorawan_mlme_confirm(gnrc_lorawan_t *mac, mlme_confirm_t *confirm)
 
 static inline void _set_be_addr(gnrc_lorawan_t *mac, uint8_t *be_addr)
 {
-    uint32_t tmp = *((uint32_t*) be_addr);
+    uint32_t tmp;
+    memcpy(&tmp, be_addr, sizeof(uint32_t));
     tmp = byteorder_swapl(tmp);
```

Comment:
I'd prefer if a 32-bit variant of `static inline uint16_t byteorder_bebuftohs(const uint8_t *buf);` were added to `core/include/byteorder.h` for that and used here.